tokio/src/time/driver/entry.rs

```diff
@@ -437,6 +437,17 @@ impl TimerShared {
         true_when
     }
 
+    /// Sets the cached time-of-expiration value.
+    ///
+    /// SAFETY: Must be called with the driver lock held, and when this entry is
+    /// not in any timer wheel lists.
+    pub(super) unsafe fn set_cached_when(&self, when: u64) {
```

Comment:
this is now only called by `mark_pending`, so it doesn't need to be pub
```suggestion
    unsafe fn set_cached_when(&self, when: u64) {
```